site/docs/about/glossary.mdx

```diff
@@ -36,3 +36,11 @@ Salt is the J.P. Morgan design system, an open-source solution for building exce
 Salt is under active development. At this stage, we don't recommend theming unless absolutely necessary.
 
 For an overview of theming, refer to our [theming documentation](/salt/themes/index).
+
+## Disabled state
+
+A component in a disabled state isn't considered relevant to the user’s workflow, as the user doesn’t need to know the component is there. It doesn't need to meet minimum contrast requirements, and a progressive reveal might be a better UI pattern. While buttons can use the `focusableWhenDisabled` prop, the system doesn't support status messages for disabled components. If a component is relevant, consider using a read-only state instead.
```

Comment:
Per @joshwooding feedback, have suggested alternative wording. Please alter as needed. Might want to revisit the read-only text as well.

A component in a disabled state isn't considered relevant to the user’s workflow _at that time_. As the function is not available to the user, the component doesn't need to meet minimum contrast requirements. Consider if a disabled state is the right UI pattern, or if a progressive reveal might be more appropriate. Salt doesn't provide error, warning or success states for disabled components–the status messaging implies a user action is required and needs to be addressed. In this case consider using a read-only state instead. Some components may make individual exceptions, Button has a `focusableWhenDisabled` prop so it can be discovered where context is important.

Reply:
Is it worth highlighting the focusableWhenDisabled also allows a tooltip to be used to explain why the action is disabled?

Reply:
bumping  the above... we should look at before merge

Reply:
Is the key differentiator focus?
Should the glossary be a succinct explanation, rather than suggesting alternative patterns, progressive reveal and focusWhenDisabled

Otherwise we may need another glossary item for progressive reveal or links to props.

This was LLM generated but you get the gist.

Read-Only

A read-only state in UI components allows users to view and interact with the content without modifying it. Users can select and copy text from read-only components, making it possible to use the information elsewhere. This state is typically used for displaying information that should not be altered, such as view-only fields or summaries. Visual indicators may be used to signal that the component is not editable, while still allowing for actions like text selection and copying.

Disabled

A disabled state in UI components renders them completely inactive and non-interactive. Users cannot focus on, select, or interact with disabled components, which includes the inability to copy text from them. This state is often used to indicate that an action is temporarily unavailable or not applicable, such as a button that is inactive until certain conditions are met. Disabled components are typically styled with visual cues, such as grayed-out text or reduced opacity, to signal their inactive status.

Reply:
Incorporating all of these versions to make a high level definition. For future consideration, do we want to include props in the glossary? How detailed / advanced should definitions in the glossary be? 

Currently I notice that the glossary provides a foundational understanding of the terms. Should users reference additional documentation if they're seeking more detailed information? If not, we may consider adding more terms here such as progressive reveal or RAG state